src/Network/Transport/Internal.hs

```diff
@@ -123,8 +124,25 @@ decodeNum16 :: Num a => ByteString -> a
 decodeNum16 = fromIntegral . decodeWord16
 
 -- | Prepend a list of bytestrings with their total length
+--   Will be an exception in case of overflow: the sum of the lengths of
+--   the ByteStrings overflows Int, or that sum overflows Word32.
 prependLength :: [ByteString] -> [ByteString]
-prependLength bss = encodeWord32 (fromIntegral . sum . map BS.length $ bss) : bss
+prependLength bss = case word32Length of
+  Nothing -> overflow
+  Just w32 -> encodeWord32 w32 : bss
+  where
```

Comment:
Code style: can we follow the typical style of indent in the body and 4 after where, ie

    f x =
        blah x
      where
        blah x = ...